src/audio/drc/drc.c

```diff
@@ -156,19 +156,17 @@ static struct comp_dev *drc_new(const struct comp_driver *drv,
 	if (bs > SOF_DRC_MAX_SIZE) {
 		comp_cl_err(&comp_drc, "drc_new(), error: configuration blob size = %u > %d",
 			    bs, SOF_DRC_MAX_SIZE);
-		return NULL;
+		goto fail;
```

Comment:
sorry for not being clear, but first alloc can stay as a return since it does not have to free anything else.